src/thunderbird_accounts/mail/tests.py

```diff
@@ -82,29 +71,47 @@ def test_sign_up_invalid_domain(self):
     def test_sign_up_already_have_account(self):
       self.c.force_login(self.user)
 
-      resp = self.c.post(reverse('sign_up_submit'), data={
-            'email_address': 'new-thundermail-address',
-            'email_domain': 'tb.pro',
-            'app_password': 'password',
-        })
+      resp = self.c.post(reverse('sign_up_submit'), data=self.sign_up_data)
 
       # expect redirect to connection-info after email created successfully
       self.assertEqual(resp.status_code, 302)
       self.assertEqual(resp.url, '/self-serve/connection-info')
 
       # now attempt to sign up for email again
-      resp = self.c.post(reverse('sign_up_submit'), data={
-            'email_address': 'new-thundermail-address',
-            'email_domain': 'tb.pro',
-            'app_password': 'password',
-        })
+      resp = self.c.post(reverse('sign_up_submit'), data=self.sign_up_data)
 
       # expect redirect back to /sign-up since already signed up
       self.assertEqual(resp.status_code, 302)
       self.assertEqual(resp.url, '/sign-up/')
 
     def test_sign_up_submit_no_user(self):
       resp = self.c.post(reverse('sign_up_submit'))
+
       # expect no user redirect to root b/c no user
       self.assertEqual(resp.status_code, 302)
       self.assertEqual(resp.url, '/')
+
+    def test_sign_up_email_already_taken(self):
+      self.c.force_login(self.user)
+
+      resp = self.c.post(reverse('sign_up_submit'), data=self.sign_up_data)
+
+      # expect redirect to connection-info after email created successfully
+      self.assertEqual(resp.status_code, 302)
+      self.assertEqual(resp.url, '/self-serve/connection-info')
+
+      # now create a 2nd user
+      second_client = Client(enforce_csrf_checks=False)
+      self.second_user = self.UserModel.objects.create(
+        fxa_id='abc456',
+        email='second-user@test.com',
+        username='Second User'
+      )
+      second_client.force_login(self.second_user)
+
+      # then have the 2nd user attempt to sign up for email using the same email address
+      resp = second_client.post(reverse('sign_up_submit'), data=self.sign_up_data)
+
+      # expect redirect back to /sign-up since sign up should fail b/c requested email address is not unique
+      self.assertEqual(resp.status_code, 302)
+      self.assertEqual(resp.url, '/sign-up/')
```

Comment:
Can you add some assertions to ensure the email account, and address aren't created in the db side too?

Reply:
Oh great idea, will do!